### PR TITLE
introduces perftools / JSONStore to improve site performance

### PR DESCRIPTION
**Pull request by owocki**
##### Description

introduces perftools / JSONStore to improve site performance

also solves gas station memory error by storing gas history objects in the database, refreshed with a management command, instead of lazily loading them on the fly and caching them in redis.

views cleaned up:
- gas/history
- all d3_views that are publicly facing
- all /results views

hopefully helps with site stability too as these views are one of the long running gunicorn requests we've seen

##### Checklist

- [x] linter status: 100% pass
- [x] changes don't break existing behavior
- [x] commit message follows [commit guidelines](https://docs.gitcoin.co/mk_contributors/#step-4-commit)

##### Affected core subsystem(s)
gas station

##### Testing

tested on my local

##### Refers/Fixes

https://sentry.gitcoin.co/gitcoin/gitcoin-prod-backend/issues/1969/?referrer=slack

### CODE REVIEW

File: app/gas/management/commands/create_gas_history.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        with transaction.atomic():
+            for breakdown in breakdowns:
+                GasHistory.objects.filter(key__startswith=breakdown).all().delete()
+                for i, __ in lines.items():
+                    view = 'gas_history'
+                    key = f"{breakdown}:{i}"
+                    print(f"- executing {breakdown} {key}")
+                    data = gas_history(breakdown, i)
+                    print("- saving")
+                    GasHistory.objects.create(
```
Comment (mbeacom) [inline app/gas/management/commands/create_gas_history.py]: Maybe we can adjust this in the future to use `.bulk_create` - quick a dirty way to do this for now would be to add an empty list, drop `.objects.create`, append to the list, and `GasHistory.objects.bulk_create(list_of_gas_history_objs)`

File: app/gas/management/commands/create_gas_history.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+'''
```
Comment (mbeacom) [inline app/gas/management/commands/create_gas_history.py]: Multi-line spanning with `'''` can cause problems.  Moving forward, can we use `"""` to span lines?

File: app/dashboard/gas_views.py [hunk 3]
```diff
@@ -235,13 +220,16 @@ def gas_guzzler_view(request):
     return TemplateResponse(request, 'gas_guzzler.html', context)
 
 
-@cached_view(timeout=60*16)
```
[the remark this answers is not in the document]
Comment (owocki) [reply app/dashboard/gas_views.py]: 1280df2ffaeb1e549b816acca8d878d852ef037d

File: app/gas/management/commands/create_gas_history.py (hunk 2 above)
Comment (owocki) [reply app/gas/management/commands/create_gas_history.py]: sure, but we should backport all the other views that have this style of top level comment in it.  this was pasted from another mgmt command

File: app/gas/management/commands/create_gas_history.py (hunk 1 above)
Comment (owocki) [reply app/gas/management/commands/create_gas_history.py]: we only create 7 rows at a time right now.  i def agree that bulk create is better tho.

afc701c17e987afeffb7dede1193d291ffa185db

**Comment by mbeacom** [comment]
from @owocki 
```
what do you think about renaming the new `GasHistory` channel to be broader..  so we can DBCache other things in it?
and maybe moving it to a new app
we could then cache the `/results` page in there, and any other non-gas related info
```
I think moving this to be more modular makes sense!

Should we make the adjustments in this branch?

**Comment by owocki** [comment]
5aa34dc2945b1f45dad212cd99ae9d4e35bbdee2 <= /results page is SO FAST now!

**Comment by owocki** [comment]
@mbeacom didnt you post a report of the slowest views we have on the site before? i cant find it..   mind linking it?

**Comment by owocki** [comment]
deployed.. site is so fast on these pages now!

> @mbeacom didnt you post a report of the slowest views we have on the site before? i cant find it.. mind linking it?

we should figure out if theres any other pages to cleanup :)

**Pull request merged**